x-pack/plugins/enterprise_search/public/applications/app_search/components/engines/engines_logic.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { HttpLogic } from '../../../shared/http';
+
+import { EngineDetails } from '../engine/types';
+
+interface EnginesValues {
+  dataLoading: boolean;
+  engines: EngineDetails[];
+  enginesTotal: number;
+  enginesPage: number;
+  metaEngines: EngineDetails[];
+  metaEnginesTotal: number;
+  metaEnginesPage: number;
+}
+
+interface OnEnginesFetch {
+  engines: EngineDetails[];
+  total: number;
+}
+interface EnginesActions {
+  onEnginesFetch({ engines, total }: OnEnginesFetch): OnEnginesFetch;
+  onMetaEnginesFetch({ engines, total }: OnEnginesFetch): OnEnginesFetch;
+  onEnginesPagination(page: number): { page: number };
+  onMetaEnginesPagination(page: number): { page: number };
+  fetchEngines(): void;
+  fetchMetaEngines(): void;
+}
+
+export const EnginesLogic = kea<MakeLogicType<EnginesValues, EnginesActions>>({
+  path: ['enterprise_search', 'app_search', 'engines_logic'],
+  actions: {
+    onEnginesFetch: ({ engines, total }) => ({ engines, total }),
+    onMetaEnginesFetch: ({ engines, total }) => ({ engines, total }),
+    onEnginesPagination: (page) => ({ page }),
+    onMetaEnginesPagination: (page) => ({ page }),
+    fetchEngines: true,
+    fetchMetaEngines: true,
+  },
+  reducers: {
+    dataLoading: [
+      true,
+      {
+        onEnginesFetch: () => false,
+      },
+    ],
```

Comment:
I'm glad this has ended up in the kea logic now.  We've discussed elsewhere that I'm not a _huge_ fan of `dataLoading`/`isLoading` props, and prefer using `null` defaults for values we load from elsewhere, and doing explicit null checks for those values. I think that is a "typescript"-er way of doing it. Not a blocker for this PR, but wanted to mention my perspective on this again.

Reply:
👍 I don't have super strong feelings one way or another, either an `isLoading` prop of some kind of a falsey check is fine/reads the same to me. What I do have stronger feelings on is preferring to remain consistent across the codebase rather than have several different / confusing instances of how we handle loading state, so if possible I'd like to tackle all instances of `dataLoading` in one fell swoop post-MVP.

Reply:
I really like the explicitness of `dataLoading`. It *could* be a selector that checks for certain values to be null.